tests/5.0/target/test_target_device.F90

```diff
@@ -0,0 +1,99 @@
+!===--- test_target_device.F90 -------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This test checks the target construct with device clause where device-
+! modifier is either ancestor or device_num. If no device_modifier is 
+! present, the behavior is the same as if device_num were present.
+! 
+!//===----------------------------------------------------------------------===//
+#define OMPVV_MODULE_REQUIRES_LINE !$omp requires reverse_offload
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_target_device
+  ! Required for device(ancestor: 1)
+  !$omp requires reverse_offload
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+
+  OMPVV_TEST_OFFLOADING
+
+  OMPVV_TEST_VERBOSE(target_device_ancestor() .ne. 0)
+  OMPVV_TEST_VERBOSE(target_device_device_num() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION target_device_ancestor()
+    INTEGER :: errors, i, which_device
+    INTEGER, DIMENSION(N) :: a
+    LOGICAL :: is_shared_env
+
+    errors = 0
+    which_device = 0
+    is_shared_env = .false.
+
+    DO i = 1, N
+       a(i) = i
+    END DO
+
+    OMPVV_TEST_AND_SET(errors, omp_get_num_devices() .le. 0) 
+    OMPVV_WARNING_IF(omp_get_num_devices() .le. 0, "[SKIPPED] Since no target devices were found, this test will be skipped.")
```

Comment:
The message is not quite true: It is not skipped but it is just testing not as much. _(And it will fail – as the host is a shared-memory device.)_